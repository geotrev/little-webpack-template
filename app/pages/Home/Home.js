import React from "react"
import ScrollUpOnMount from "app/components/ScrollUpOnMount"

export default function Home() {
  return (
    <div id="home">
      <ScrollUpOnMount />
      <span role="img" aria-label="waving emoji" className="waving emoji">
        ðŸ‘‹
      </span>
      <h1>{"Hello there!"}</h1>
      <p>{"This is a React starter kit built with Webpack."}</p>
    </div>
  )
}
