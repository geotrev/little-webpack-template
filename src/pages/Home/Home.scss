#home {
  flex: 1 1 auto;
  max-width: 100%;
  display: flex;
  align-items: center;
  flex-flow: column nowrap;
  font-family: monospace;
  font-size: 16px;
  text-align: center;
  padding: 120px 20px;

  .waving {
    animation: wave 0.75s alternate infinite;
  }

  h1 {
    color: #15b78f;
    font-weight: bold;
    font-size: 3rem;
  }

  p {
    font-size: 1.5rem;
    color: white;
  }
}

@keyframes wave {
  100% {
    transform: rotate(20deg);
  }
}
