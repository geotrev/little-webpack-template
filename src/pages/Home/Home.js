import React from 'react';
import './Home.scss';
import { ScrollUpOnMount } from 'helpers';

import Intro from './Intro/Intro';

const Home = () => {
  return (
    <div id="home">
      <ScrollUpOnMount />
      <div className="emoji">ğŸ‘‹</div>
      <Intro greeting="This is a boilerplate built with webpack and React." />
    </div>
  );
}

export default Home;
