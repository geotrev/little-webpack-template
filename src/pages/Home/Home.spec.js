import Home from "./Home"

describe("<Home />", () => {
  function mountComponent() {
    return mount(<Home />)
  }

  let wrapper

  beforeEach(() => {
    wrapper = mountComponent()
  })

  it("renders", () => {
    expect(wrapper).to.exist
    expect(wrapper).to.have.length(1)
  })

  it("renders an h1", () => {
    expect(wrapper).to.have.descendants("h1")
  })

  it("renders an .emoji", () => {
    expect(wrapper).to.have.descendants(".emoji")
  })
})
