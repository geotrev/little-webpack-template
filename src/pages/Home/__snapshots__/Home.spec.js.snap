// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> matches snapshot 1`] = `
<Home>
  <div
    id="home"
  >
    <ScrollUpOnMount />
    <div
      className="waving emoji"
    >
      ðŸ‘‹
    </div>
    <h1>
      Hello there!
    </h1>
    <p>
      This is a React starter kit built with Webpack.
    </p>
  </div>
</Home>
`;
